Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -376,12 +386,19 @@ private async Task<HttpResponseMessage> PushDeployAsync(ArtifactDeploymentInfo d
 
                     if (ArmUtils.IsArmRequest(Request))
                     {
+                        _tracer.Trace("Reading the artifact URL from the ARM request JSON");
                         requestObject = requestObject.Value<JObject>("properties");
                     }
 
                     var packageUri = requestObject.Value<string>("packageUri");
+                    if (string.IsNullOrEmpty(packageUri))
+                    {
+                        _tracer.Trace("PackageUri was empty in the JSON request");
+                        throw new ArgumentException($"PackageUri was empty in the JSON request");
```

Comment:
What Http Status the client gets when we throw `ArgumentException`? Is it 4xx or 5xx?

Reply:
It will be 500 with clear exception message about the URL.
"ExceptionMessage": "PackageUri was empty in the JSON request"

Reply:
We should be returning 400. Missing `packageUrl` means it is a bad request and fault from the client side if I understand correctly. I will let Suwat have a say on this one as we are returning `ArgumentException` in other places in the same method too.

Reply:
@khkh-ms Correction: It is already sending 400 Bad Request with ExceptionMessage